cfme/infrastructure/cluster.py

```diff
@@ -117,7 +117,12 @@ class All(CFMENavigateStep):
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
     def step(self):
-        self.parent_view.navigation.select('Compute', 'Infrastructure', 'Clusters')
+        try:
```

Comment:
Noted, I would like to see that logic here instead of the try/except. `self.obj.provider` should be accessible to base the navigation off of the provider type.